integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/distrox/DistroXEncryptedVolumeTest.java

```diff
@@ -200,6 +200,7 @@ public void testCreateDistroXWithEncryptedVolumesInSingleRG(TestContext testCont
 
         testContext
                 .given(EnvironmentNetworkTestDto.class)
+                    .withServiceEndpoints()
```

Comment:
Unfortunately I got error: 
```
11:20:42.324 [TestNG-test=azure_e2e_tests-1] ERROR c.s.i.cloudbreak.context.TestContext [DistroXEncryptedVolumeTest.testCreateDistroXWithEncryptedVolumesInSingleRG] - then [DistroXEncryptedVolumeTest$$Lambda$940/0x0000000801708400] assertion is failed: Private DNS zone id for database was requested to be /subscriptions/78fcb648-af6f-4eba-9db2-1a57c973cb7b/resourceGroups/rg-gpapp-dns-zone/providers/Microsoft.Network/privateDnsZones/privatelink.postgres.database.azure.com, but is null, name: azure-test-e44fc1f399834c75b
com.sequenceiq.it.cloudbreak.exception.TestFailException: Private DNS zone id for database was requested to be /subscriptions/78fcb648-af6f-4eba-9db2-1a57c973cb7b/resourceGroups/rg-gpapp-dns-zone/providers/Microsoft.Network/privateDnsZones/privatelink.postgres.database.azure.com, but is null
	at com.sequenceiq.it.cloudbreak.cloud.v4.azure.AzureCloudProvider.assertServiceEndpoint(AzureCloudProvider.java:163)
	at com.sequenceiq.it.cloudbreak.cloud.v4.CloudProviderProxy.assertServiceEndpoint(CloudProviderProxy.java:402)
	at com.sequenceiq.it.cloudbreak.testcase.e2e.distrox.DistroXEncryptedVolumeTest.lambda$testCreateDistroXWithEncryptedVolumesInSingleRG$0(DistroXEncryptedVolumeTest.java:216)
	at com.sequenceiq.it.cloudbreak.context.TestContext.then(TestContext.java:382)
	at com.sequenceiq.it.cloudbreak.dto.AbstractEnvironmentTestDto.then(AbstractEnvironmentTestDto.java:97)
	at com.sequenceiq.it.cloudbreak.dto.AbstractEnvironmentTestDto.then(AbstractEnvironmentTestDto.java:92)
	at com.sequenceiq.it.cloudbreak.testcase.e2e.distrox.DistroXEncryptedVolumeTest.testCreateDistroXWithEncryptedVolumesInSingleRG(DistroXEncryptedVolumeTest.java:215)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
	at org.testng.internal.MethodInvocationHelper$1.runTestMethod(MethodInvocationHelper.java:238)
	at org.springframework.test.context.testng.AbstractTestNGSpringContextTests.run(AbstractTestNGSpringContextTests.java:184)
	at org.testng.internal.MethodInvocationHelper.invokeHookable(MethodInvocationHelper.java:252)
	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:47)
	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
	at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:33)
	at org.testng.internal.MethodInvocationHelper.invokeWithTimeoutWithNoExecutor(MethodInvocationHelper.java:314)
	at org.testng.internal.MethodInvocationHelper.invokeWithTimeout(MethodInvocationHelper.java:280)
	at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:605)
	at org.testng.internal.TestInvoker.invokeTestMethod(TestInvoker.java:174)
	at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:46)
	at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:822)
	at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:147)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
```
Furthermore from the `"serviceEndpointCreation": "DISABLED",` in the `POST http://qa-cloud-canary.eng.hortonworks.com/environmentservice/api/v1/env`. So the `databasePrivateDnsZoneId` and its value is missing from the response. This is the cause of the above ERROR.

The root cause of the issue is:
```
...
testContext
                .given(EnvironmentNetworkTestDto.class)
                    .withServiceEndpoints()
                .given("telemetry", TelemetryTestDto.class)
                    .withLogging()
                    .withReportClusterLogs()
                .given(EnvironmentTestDto.class)
                    .withNetwork()
                    .withResourceGroup(ResourceGroupTest.AZURE_RESOURCE_GROUP_USAGE_SINGLE, resourceGroupForTest)
                    .withResourceEncryption()
                    .withTelemetry("telemetry")
                    .withTunnel(Tunnel.CLUSTER_PROXY)
                    .withCreateFreeIpa(Boolean.FALSE)
...
```
So the `ResourceGroupTest.AZURE_RESOURCE_GROUP_USAGE_SINGLE` setting has been applied later than the `withServiceEndpoints()` try to check this. So here the default value is going to be applied:
```
private String getResourceGroupUsage() {
        return getTestParameter().get(ResourceGroupTest.AZURE_RESOURCE_GROUP_USAGE);
}
```
I think here we should set this as:
```
.given(EnvironmentNetworkTestDto.class)
                    .withServiceEndpoints(ServiceEndpointCreation.ENABLED_PRIVATE_ENDPOINT)
...
```
So we should introduce an additional method to the `EnvironmentNetworkTestDto` for this case:
```
public EnvironmentNetworkTestDto withServiceEndpoints(ServiceEndpointCreation serviceEndpointCreation) {
        getRequest().setServiceEndpointCreation(serviceEndpointCreation);
        return this;
}
```

Reply:
You are right!

As it turned out, I had an extra setting in my local app.yml:
`resourcegroup:
      usage: SINGLE_WITH_DEDICATED_STORAGE_ACCOUNT
      name: rg-gpapp-single-rg`

This is an old setting added when single RG support was implemented, but was never used in any job.
This will turn on the use of single RG and hence the private endpoint as it is coupled to that.

If you add that to the test then it should run through smoothly.

Reply:
@aszegedi as agreed I tested my PR with different settings:
- one for private endpoint and specifying a private DNS zone, and
- one for service endpoints

Both tests could be started successfully (I checked the env request in env service) and were running (I waited it got past freeipa creation).
I used the following settings:

**Private endpoints:**
```
    databasePrivateDnsZoneId: ....

    resourcegroup:
      usage: SINGLE_WITH_DEDICATED_STORAGE_ACCOUNT
      name: rg-gpapp-single-rg

```

**Service endpoints:**
  ```
    databasePrivateDnsZoneId: 
    resourcegroup:
        usage: MULTIPLE
        name: 
```